ray-operator/controllers/ray/utils/util.go

```diff
@@ -71,6 +71,33 @@ func IsCreated(pod *corev1.Pod) bool {
 	return pod.Status.Phase != ""
 }
 
+// CheckRayHeadRunningAndReady returns (isRayHeadRunning, reason, message) tuple.
+func CheckRayHeadRunningAndReady(ctx context.Context, pods corev1.PodList) (bool, string, string) {
+	log := ctrl.LoggerFrom(ctx)
+	headPodFound := false
+	for _, pod := range pods.Items {
+		if pod.Labels[RayNodeTypeLabelKey] == string(rayv1.HeadNode) {
+			headPodFound = true
+			if pod.Status.Phase != corev1.PodRunning {
+				log.Info(fmt.Sprintf("Head pod is not running; Pod Name: %s; Pod Status.Phase: %v", pod.Name, pod.Status.Phase))
+				return false, "HeadPodNotRunning", "Head pod is not running"
+			}
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodReady && cond.Status != corev1.ConditionTrue {
```

Comment:
Could we use the `utils.IsRunningAndReady` here?

Reply:
my thought was that `utils.IsRunningAndReady` only give us the information about Pod phase is Running + Pod is ready, it didn't tell us more fine-grain like whether it's due to Pod not running yet or Pod not ready yet (raylet may not pass readiness check yet) so i wrote the own checking